lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -340,11 +340,11 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
-  test "unused guard" do
+  test "same result / always matches" do
```

Comment:
Let's leave it as is. The test title change is kinda unrelated to this patch.

Btw, the reason it is called unused guard is because the compiler knows the guard will never be used. So it is also technically correct. :)